src/VisualStudio/CSharp/Impl/Options/CSharpVisualStudioCopilotOptionsService.cs

```diff
@@ -79,11 +77,21 @@ private async Task<bool> IsCopilotOptionEnabledAsync(CopilotOption option)
         if (!IsGithubCopilotLoadedAndSignedIn)
             return false;
 
-        var settingManager = await _settingsManagerTask.ConfigureAwait(false);
-        // The bool setting is persisted as 0=None, 1=True, 2=False, so it needs to be retrieved as an int.
-        // If isEnabled is 0 or the value is not persisted, we should return the default value for the option.
-        var isEnabled = settingManager.GetValueOrDefault($"{CopilotOptionNamePrefix}.{option.Name}", 0);
-        return isEnabled == 1 || (isEnabled == 0 && option.DefaultValue);
+        if (_settingsReader == null)
+        {
+            var settingsManager = await _settingsManagerTask.ConfigureAwait(false);
+            _settingsReader = settingsManager.GetReader();
+        }
+
+        try
+        {
+            var setting = _settingsReader.GetValue<bool>($"{option.Name}");
+            return setting.Value;
```

Comment:
Should you be verifying that the `setting.Outcome` was `Success` and returning `option.DefaultValue` if not?

Reply:
Maybe something like
```cs
if (_settingsReader.GetValue<bool>(option.Name) is { Outcome: SettingRetrievalOutcome.Success } setting)
    return setting.Value;

return option.DefaultValue;
```